### DISCUSSION TITLE
Add Client-Side Window Decorations for GNOME Wayland Compatibility

### DISCUSSION THREAD

**Discussion by first-storm** [ppy/osu#28207] [Q&A] [answered]
Description

I am encountering an issue with osu! when running it on GNOME Wayland. The application does not display window decorations, which makes it difficult to move, resize, or close the window using standard methods.
Background

GNOME Wayland supports only client-side window decorations (CSD). This means that applications are responsible for drawing their own window decorations. Currently, osu! does not seem to implement client-side window decorations, leading to a lack of window controls when running in this environment.
Expected Behavior

When osu! is running in a GNOME Wayland session, it should detect this environment and draw its own window decorations. This will ensure that users can interact with the window as expected.
Steps to Reproduce

    Run a GNOME session with Wayland.
    Launch osu!.
    Observe the lack of window decorations.

Environment

    Operating System: openSUSE Tumbleweed 20240519
    GNOME Version: 46.0
    osu! Version: [2024.412.1](https://github.com/ppy/osu/releases/tag/2024.412.1)

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/discussions/16693
